frontend/components/SetupPage/SetupWelcome.tsx

```diff
@@ -129,9 +130,15 @@ export const SetupWelcomeLogin = () => {
   const { goto } = useSetup();
   const { goto: gotoPage } = usePageState();
 
-  const { masterSafeAddress, masterWallets: wallets } =
-    useMasterWalletContext();
-  const { isBalanceLoaded, masterEoaBalance: eoaBalance } = useBalanceContext();
+  const { selectedService } = useServices();
+  const { masterSafes, masterWallets: wallets } = useMasterWalletContext();
+  const { isLoaded, walletBalances } = useBalanceContext();
+
+  const masterSafe =
+    masterSafes?.find(
+      (safe) => safe.chainId === selectedService?.home_chain_id,
+    ) ?? null;
+  const eoaBalanceEth = walletBalances?.find((balance) => balance.isNative);
```

Comment:
unsure it shouldn't check chainId as well, and also maybe better use `useMasterBalances`, worth checking with @truemiller

Reply:
yes, we should, have suggested TODO comment here: https://github.com/valory-xyz/olas-operate-app/pull/478#discussion_r1852704485

not a major issue for getting predict only version for now though